includes/admin/class-wp-job-manager-promoted-jobs-admin.php

```diff
@@ -357,17 +364,34 @@ public function get_promote_jobs_template() {
 	public function promoted_jobs_admin_footer() {
 		$screen = get_current_screen();
 
-		if ( in_array( $screen->id, [ 'edit-job_listing', 'job_listing' ], true ) ) { // Job listing and job editor.
+		// Job editor.
+		if ( 'job_listing' === $screen->id && ! \WP_Job_Manager_Admin_Notices::is_dismissed( self::JOB_EDITOR_MODAL_NOTICE, true ) ) {
+
+			$notice_wrapper_attributes = \WP_Job_Manager_Admin_Notices::get_dismissible_notice_wrapper_attributes( self::JOB_EDITOR_MODAL_NOTICE );
+
+			wp_enqueue_script( 'job_manager_notice_dismiss' );
+
 			?>
 			<template id="promote-job-template">
 				<?php echo $this->get_promote_jobs_template(); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>
 			</template>
-			<dialog class="wpjm-dialog" id="promote-dialog"></dialog>
+			<dialog
+				class="wpjm-dialog wpjm-admin-modal-notice is-dismissible"
+				id="promote-dialog"
+				<?php echo $notice_wrapper_attributes; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>
+			></dialog>
 			<?php
 		}
 
-		if ( 'edit-job_listing' === $screen->id ) { // Job listing.
+		// Job listing.
+		if ( 'edit-job_listing' === $screen->id ) {
+
 			?>
+			<template id="promote-job-template">
```

Comment:
I am probably missing something but do we need this? It didn't exist before.

Reply:
It was output from the first `if` before (`if ( in_array( $screen->id, [ 'edit-job_listing', 'job_listing' ], true ) )`), but now the two screens are a bit two different so broke it up.

The root of all this is that the same modal is used for the editor popup dialog and the job list 'Promote' buttons.

Reply:
Ah missed the change in the if! Thanks!